core/src/main/java/org/apache/calcite/rel/core/RepeatUnion.java

```diff
@@ -61,12 +64,20 @@
    */
   public final int iterationLimit;
 
+  /**
+   * Transient table where repeat union's intermediate results will be stored.
+   */
+  protected final RelOptTable transientTable;
+
```

Comment:
Do we need to update the class javadoc based on this change to explain how is this table supposed to be used? Also can this be null (optional) in case some concrete implementations do not want to make use of it?